src/qml/qmlapplication.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include <coreservices.h>
+
+#include <QApplication>
+#include <QFileSystemWatcher>
+#include <QQmlApplicationEngine>
+
+namespace mixxx {
+namespace qml {
+
+class QmlApplication : public QObject {
+    Q_OBJECT
+  public:
+    QmlApplication(
+            QApplication* app,
+            std::shared_ptr<CoreServices> pCoreServices);
+    ~QmlApplication() = default;
```

Comment:
`override`

Reply:
default override? Would it be better to simply remove this line?

Reply:
No, `override` should be used for all destructors of derived classes as a convention. C++ is so error prone and requires a lot of informal boilerplate.

Reply:
If omitted the base class is not checked for a virtual destructor. QObject has one, but nevertheless follow the convention. If you inherit than add either a `virtual` or an `override` destructor.

Reply:
Fixed and squashed into the first commit.